File: CalSnap/Features/Onboarding/OnboardingStep.swift
import Foundation

enum OnboardingStep: Int, CaseIterable, Hashable {
    case welcome
    case profileSetup
    case goalSetup
    case caloriePreview
    case healthKit
    case apiKeys
    case done

    var title: String {
        switch self {
        case .welcome: String(localized: "onboarding.step.welcome")
        case .profileSetup: String(localized: "onboarding.step.profile")
        case .goalSetup: String(localized: "onboarding.step.goals")
        case .caloriePreview: String(localized: "onboarding.step.calorieTarget")
        case .healthKit: String(localized: "onboarding.step.healthKit")
        case .apiKeys: String(localized: "onboarding.step.apiKeys")
        case .done: String(localized: "onboarding.step.done")
        }
    }
}

File: calsnap-web/lib/onboarding/onboarding-step.ts
export const ONBOARDING_STEPS = [
  'welcome',
  'profileSetup',
  'goalSetup',
  'caloriePreview',
  'done',
] as const;

export type OnboardingStep = (typeof ONBOARDING_STEPS)[number];

export const ONBOARDING_STEP_TITLES: Record<OnboardingStep, string> = {
  welcome: 'Welcome',
  profileSetup: 'Your profile',
  goalSetup: 'Your goal',
  caloriePreview: 'Calorie target',
  done: 'All set',
};

export function onboardingStepIndex(step: OnboardingStep): number {
  return ONBOARDING_STEPS.indexOf(step);
}

export function onboardingProgress(step: OnboardingStep): number {
  return (onboardingStepIndex(step) + 1) / ONBOARDING_STEPS.length;
}
